include/deal.II/grid/grid_tools.h

```diff
@@ -1032,6 +1032,52 @@ namespace GridTools
 
+  /**
+   * Compute a collection of bounding boxes so that all cells for which the given predicate is true, are
+   * completely enclosed in at least one of the bounding boxes.
+   *
+   * For each cell at a given refinement level containing active cells for which predicate is true,
+   * the function creates a bounding box of its children for which predicate is true.
+   *
+   * This results in a cover of these cells; the parameters @p allow_merge and @p max_boxes are used to
+   * reduce the number of cells at a computational cost and covering a bigger n-dimensional volume.
+   *
+   * The parameters to control the algorithm are:
+   * - @p predicate : the property of the cells to enclose e.g. IteratorFilters::LocallyOwnedCell .
+   *  The predicate is tested only on active cells.
+   * - @p refinement_level : it defines the level at which the initial bounding box are created. The refinement
+   *  should be set to a coarse refinement level. If @p refinement_level is higher than the number of levels of the
+   *  triangulation an exception is thrown.
+   * - @p allow_merge : This flag allows for box merging and, by default, is false. The algorithm has a cost of
+   *  O(N^2) where N is the number of the bounding boxes created from the refinement level; for this reason, if
+   *  the flag is set to true, make sure to choose wisely a coarse enough @p refinement_level.
+   * - @p max_boxes : the maximum number of bounding boxes to compute. If more are created the smaller ones are
+   *  merged with neighbors. By default after merging the boxes which can be expressed as a single one no
+   *  more boxes are merged. See the BoundingBox::get_neighbor_type () function for details.
+   *  Notice only neighboring cells are merged (see the @p get_neighbor_type  function in bounding box class): if
+   *  the target number of bounding boxes max_boxes can't be reached by merging neighbors an exception is thrown
+   *
+   * The following image describes an example of the algorithm with @p refinement_level = 2, @p allow_merge = true
+   * and @p max_boxes = 1. The cells with the property predicate are in red, the area of a bounding box is
+   * slightly orange.
+   * @image html bounding_box_predicate.png
+   * - 1. In black we can see the cells of the current level.
+   * - 2. For each cell containing the red area a bounding box is created: by default these are returned.
+   * - 3. Because @p allow_merge = true the number of bounding boxes is reduced while not changing the cover.
+   *  If @p max_boxes was left as default or bigger than 1 these two boxes would be returned.
+   * - 4. Because @p max_boxes = 1 the smallest box is merged to the bigger one.
+   * Notice it is important to choose the parameters wisely. For instance, @p allow_merge = false and
+   * @p refinement_level = 1 returns the very same bouding box but with a fraction of the computational cost.
```

Comment:
typo "bouding box"

Reply:
Thank you very much @tjhei : I just saw the comments (this weekend I was busy trying to make MPI work) now I shall update everything.

Now that you point it out finding all children of a cell is quicker. I was thinking about something like:
- new_cell = cell-> parent()
- check if all children are "locally owned"
- if so and cell->level < refinement_level then new_cell = new_cell -> parent()
- check all the other children: if all locally owned new_cell = new_cell -> parent
and then I could also do some tricks with merging (following this algorithm you can "avoid" merging one by one all the cells)

The problem is that after this I might still want to run the heavy algorithm because there might still be "mergeable" bounding boxes which are not children of the same cell. Anyway it seems a good improvement, I shall add it asap

Reply:
> I shall add it asap

We could also wait and do it in a followup PR. I am just throwing ideas/comments at you so I can understand things better.